test/groovy/ChecksPublishResultsTest.groovy

```diff
@@ -0,0 +1,269 @@
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.Ignore
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import static org.junit.Assert.assertEquals
+import static org.junit.Assert.assertTrue
+
+import util.Rules
+
+class ChecksPublishResultsTest extends BasePipelineTest {
+    Map publisherStepOptions
+    List archiveStepPatterns
+
+    @Rule
+    public RuleChain ruleChain = Rules.getCommonRules(this)
+
+    def stepUnderTest
```

Comment:
we should name it like `checksPublishResultsScript` as we do in other steps as well